storage/src/vespa/storage/distributor/operations/idealstate/idealstateoperation.cpp

```diff
@@ -177,45 +185,61 @@ class RequestBucketInfoChecker : public PendingMessageTracker::Checker
 }
 
 bool
-IdealStateOperation::checkBlock(const document::BucketId& bId,
+IdealStateOperation::checkBlock(const document::Bucket &bucket,
                                 const PendingMessageTracker& tracker) const
 {
     IdealStateOpChecker ichk(*this);
     RequestBucketInfoChecker rchk;
     const std::vector<uint16_t>& nodes(getNodes());
     for (size_t i = 0; i < nodes.size(); ++i) {
-        tracker.checkPendingMessages(nodes[i], bId, ichk);
+        tracker.checkPendingMessages(nodes[i], bucket, ichk);
         if (ichk.blocked) {
             return true;
         }
         // Check messages sent to null-bucket (i.e. any bucket) for the node.
-        tracker.checkPendingMessages(nodes[i], document::BucketId(), rchk);
+        document::Bucket nullBucket(bucket.getBucketSpace(), document::BucketId());
+        tracker.checkPendingMessages(nodes[i], nullBucket, rchk);
         if (rchk.blocked) {
             return true;
         }
+        if (bucket.getBucketSpace() != BucketSpace::placeHolder()) {
+            nullBucket = document::Bucket(BucketSpace::placeHolder(), document::BucketId());
+            tracker.checkPendingMessages(nodes[i], nullBucket, rchk);
+            if (rchk.blocked) {
+                return true;
+            }
+        }
     }
     return false;
 }
 
 bool
 IdealStateOperation::checkBlockForAllNodes(
-        const document::BucketId& bid,
+        const document::Bucket &bucket,
         const PendingMessageTracker& tracker) const
 {
     IdealStateOpChecker ichk(*this);
     // Check messages sent to _any node_ for _this_ particular bucket.
-    tracker.checkPendingMessages(bid, ichk);
+    tracker.checkPendingMessages(bucket, ichk);
     if (ichk.blocked) {
         return true;
     }
     RequestBucketInfoChecker rchk;
     // Check messages sent to null-bucket (i.e. _any bucket_) for the node.
     const std::vector<uint16_t>& nodes(getNodes());
     for (size_t i = 0; i < nodes.size(); ++i) {
-        tracker.checkPendingMessages(nodes[i], document::BucketId(), rchk);
+        document::Bucket nullBucket(bucket.getBucketSpace(), document::BucketId());
+        tracker.checkPendingMessages(nodes[i], nullBucket, rchk);
         if (rchk.blocked) {
             return true;
         }
+        if (bucket.getBucketSpace() != BucketSpace::placeHolder()) {
+            nullBucket = document::Bucket(BucketSpace::placeHolder(), document::BucketId());
+            tracker.checkPendingMessages(nodes[i], nullBucket, rchk);
+            if (rchk.blocked) {
+                return true;
+            }
+        }
```

Comment:
Looks like this block could be factored out and de-duped with the one in `checkBlock`